From 04655118a9d0ee363be228eb8bb1122f35b7e92a Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 9 Nov 2018 11:22:08 +0800
Subject: [PATCH 5079/5242] MLK-20301 gpu: imx: dpu: layerblend: Remove
 several invalid registers & wrappers

commit  9f7f845c1237ffb65b1c47f620143348b2f7f49d from
https://source.codeaurora.org/external/imx/linux-imx.git

The layerblend units don't contain the CONTROLWORD, CURPIXELCNT,
LASTPIXELCNT and PERFCOUNTER registers, so let's remove them
and their wrappers(no one is calling them), which were introduced
accidentally.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-layerblend.c |   64 ----------------------------------
 include/video/dpu.h                  |    4 ---
 2 files changed, 68 deletions(-)

diff --git a/drivers/gpu/imx/dpu/dpu-layerblend.c b/drivers/gpu/imx/dpu/dpu-layerblend.c
index 718c8db..0ca3221 100644
--- a/drivers/gpu/imx/dpu/dpu-layerblend.c
+++ b/drivers/gpu/imx/dpu/dpu-layerblend.c
@@ -71,20 +71,6 @@
 #define PRIMCONTROLWORD				0x18
 #define SECCONTROLWORD				0x1C
 
-#define CONTROLWORD				0x18
-#define CURPIXELCNT				0x1C
-static u16 get_xval(u32 pixel_cnt)
-{
-	return pixel_cnt && 0xFF;
-}
-
-static u16 get_yval(u32 pixel_cnt)
-{
-	return pixel_cnt >> 16;
-}
-#define LASTPIXELCNT				0x20
-#define PERFCOUNTER				0x24
-
 struct dpu_layerblend {
 	void __iomem *pec_base;
 	void __iomem *base;
@@ -262,56 +248,6 @@ void layerblend_position(struct dpu_layerblend *lb, int x, int y)
 }
 EXPORT_SYMBOL_GPL(layerblend_position);
 
-u32 layerblend_last_control_word(struct dpu_layerblend *lb)
-{
-	u32 val;
-
-	mutex_lock(&lb->mutex);
-	val = dpu_lb_read(lb, CONTROLWORD);
-	mutex_unlock(&lb->mutex);
-
-	return val;
-}
-EXPORT_SYMBOL_GPL(layerblend_last_control_word);
-
-void layerblend_pixel_cnt(struct dpu_layerblend *lb, u16 *x, u16 *y)
-{
-	u32 val;
-
-	mutex_lock(&lb->mutex);
-	val = dpu_lb_read(lb, CURPIXELCNT);
-	mutex_unlock(&lb->mutex);
-
-	*x = get_xval(val);
-	*y = get_yval(val);
-}
-EXPORT_SYMBOL_GPL(layerblend_pixel_cnt);
-
-void layerblend_last_pixel_cnt(struct dpu_layerblend *lb, u16 *x, u16 *y)
-{
-	u32 val;
-
-	mutex_lock(&lb->mutex);
-	val = dpu_lb_read(lb, LASTPIXELCNT);
-	mutex_unlock(&lb->mutex);
-
-	*x = get_xval(val);
-	*y = get_yval(val);
-}
-EXPORT_SYMBOL_GPL(layerblend_last_pixel_cnt);
-
-u32 layerblend_perfresult(struct dpu_layerblend *lb)
-{
-	u32 val;
-
-	mutex_lock(&lb->mutex);
-	val = dpu_lb_read(lb, PERFCOUNTER);
-	mutex_unlock(&lb->mutex);
-
-	return val;
-}
-EXPORT_SYMBOL_GPL(layerblend_perfresult);
-
 struct dpu_layerblend *dpu_lb_get(struct dpu_soc *dpu, int id)
 {
 	struct dpu_layerblend *lb;
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 42a9ff5..16adc62 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -664,10 +664,6 @@ void layerblend_pixengcfg_clken(struct dpu_layerblend *lb,
 void layerblend_control(struct dpu_layerblend *lb, lb_mode_t mode);
 void layerblend_blendcontrol(struct dpu_layerblend *lb, bool sec_from_scaler);
 void layerblend_position(struct dpu_layerblend *lb, int x, int y);
-u32 layerblend_last_control_word(struct dpu_layerblend *lb);
-void layerblend_pixel_cnt(struct dpu_layerblend *lb, u16 *x, u16 *y);
-void layerblend_last_pixel_cnt(struct dpu_layerblend *lb, u16 *x, u16 *y);
-u32 layerblend_perfresult(struct dpu_layerblend *lb);
 struct dpu_layerblend *dpu_lb_get(struct dpu_soc *dpu, int id);
 void dpu_lb_put(struct dpu_layerblend *lb);
 
-- 
1.7.9.5

